tests/sentry/sentry_apps/api/bases/test_sentryapps.py

```diff
@@ -16,64 +19,95 @@
 from sentry.testutils.cases import TestCase
 from sentry.testutils.helpers.options import override_options
 from sentry.testutils.silo import control_silo_test
+from sentry.users.models.user import User
+
+
+def skip_request_auth(
+    request: HttpRequest, user: User | None = None, token: ApiToken | None = None
+) -> Request:
+    """
+    If we try to convert an HttpRequest to a Request object with Request(), certain attributes like
+    auth won't properly be set. E.g When request.auth attempts to be created, the authenticator steps
+    will fail due to no given authenticators (or in token auth cases, improper headers).
+    This func will flag to the Request constructor that we should trust this user/token and skip the authenticator steps
+    """
+    if user:
+        request._force_auth_user = user
+
+    if token:
+        request._force_auth_token = token
```

Comment:
The 'flag' in [rest_framework](https://github.com/encode/django-rest-framework/blob/master/rest_framework/request.py#L176-L180)
Kinda jank.99 because this method will fail typing as the attributes (`_force_auth_X`) don't exist on the base `HttpRequest` object, so RIP.

There are a few other options we could take too...
- Create a dummy `Endpoint` object add the tested permission to its `permission_classes` property and then pass the HttpRequest to the endpoints `initialize_request` function which is more similar to how it works in reality but we also introduce a 'new variable' in the endpoint so not sure if that's like straying away/not really a unit test.

- Pass in some fun authenticators to the `Request` object, most of the test cases are provided a user that would work with `SessionAuthentication`. Only exception is the test which takes in no user and uses ApiToken to authenticate, we could use a different authentication and set the header. Kinda similar concern to the previous point where we have to work around another system/tie in another 'variable'

- something else ?????????? 👀

Reply:
> Make another make_request utils func that returns a Request obj ??? ( no clue what this would look like)

This approach could work well and be done in way that satisfies mypy. We could construct an instance of a relevant endpoint and use that endpoint to wrap `HttpRequest` with `Request`.

Something like

```python
 drf_request: Request = DiscordInteractionsEndpoint().initialize_request(self.request)
```

is how the discord integration is handling this scenario.